service/history/replicatorQueueProcessor.go

```diff
@@ -105,11 +111,17 @@ func newReplicatorQueueProcessor(shard ShardContext, historyCache *historyCache,
 	return processor
 }
 
-func (p *replicatorQueueProcessorImpl) process(qTask queueTaskInfo) (int, error) {
+func (p *replicatorQueueProcessorImpl) getTaskFilter() queueTaskFilter {
+	return p.replicationTaskFilter
+}
+
+func (p *replicatorQueueProcessorImpl) process(qTask queueTaskInfo, shouldProcessTask bool) (int, error) {
```

Comment:
I am confused why `shouldProcessTask` passed in here. if `! shouldProcessTask`, shouldn't we just not call process?

Reply:
`WorkflowTimeout / DeleteHistory / ArchiveHistory / CloseExecution passive tasks are guaranteed to be executed.` meaning even if the task is consider active, passive logic will still process that task